test/tap/cache-process-uptime.js

```diff
@@ -0,0 +1,67 @@
+var npm = require.resolve('../../')
+var test = require('tap').test
+var path = require('path')
+var rimraf = require('rimraf')
+var mkdirp = require('mkdirp')
+var mr = require('npm-registry-mock')
+var common = require('../common-tap.js')
+var cache = path.resolve(__dirname, 'cache-process-uptime')
+var execFile = require('child_process').execFile
+var pkg = path.resolve(__dirname, '..', 'packages', 'npm-test-test-package')
+var nm = path.resolve(pkg, 'node_modules')
+var server
+
+test('mock reg', function (t) {
+  rimraf.sync(cache)
+  mkdirp.sync(cache)
+  mr({ port: common.port }, function (er, s) {
+    server = s
+    t.pass('ok')
+    t.end()
+  })
+})
+
+test('npm install with cache-min shorter than install', function (t) {
+  rimraf.sync(nm)
+  execFile(process.execPath, [
+    npm,
+    '--cache=' + cache,
+    '--registry=' + common.registry,
+    '--loglevel=silly',
+    // has to be > 0 or caching is disabled, but less than uptime, and since
+    // node startup time is 10s of ms and this is 0.1ms, it should always be
+    // less than process.uptime()
+    '--cache-min=0.00001',
+    '--cache-max=Infinity',
+    'install', 'underscore@1.5.1'
+  ], {cwd: pkg}, function (err, stdout, stderr) {
+    t.ifErr(err, 'Should not error')
+    t.match(stderr, /^npm sill cache (hit|miss) with timeout \d/m)
+    t.notMatch(stderr, /^npm sill cache (hit|miss) with timeout 0\.00001$/m)
```

Comment:
There's a (nigh-infintesimal) chance that an ordinary run could end up with the timeout set on the CLI, but I think this is reasonable for now and we'll come up with something else if it turns out to be flappy in practice.